File: ROS1_src/learning_topic/src/pose_subscriber.cpp
//该例程将订阅/turtle1/pose话题，消息类型geometry_msgs::Twist
#include<ros/ros.h>
#include"turtlesim/Pose.h"

//消息回调函数
void poseCallback(const turtlesim::Pose::ConstPtr& msg)
{
    //打印消息
    ROS_INFO("Turtle pose:x:%0.6f,y:%0.6f",msg->x,msg->y);
}

int main(int argc,char**argv)
{
    ros::init(argc,argv,"pose_subscriber");
    ros::NodeHandle n;

    //创建一个Subscriber，订阅名为/turtlr1/pose点topic，注册回调函数poseCallback
    ros::Subscriber pose_pub=n.subscribe("/turtle1/pose",10,poseCallback);

    ros::spin(); //循环等待回调函数

    return 0;
}

File: ROS1_src/learning_topic/scripts/pose_subscriber.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
import rospy
from turtlesim.msg import Pose

def poseCallback(msg):
    rospy.loginfo("Turtle pose:x:%0.6f,y:%0.6f",msg.x,msg.y)

def pose_subscriber():
    rospy.init_node('pose_subscriber',anonymous=True)

    #创建一个Subscriber，订阅名为、turtle1/pose点topic，注册回调函数poseCallback
    rospy.Subscriber("/turtle1/pose",Pose,poseCallback)

    rospy.spin() #循环等待回调函数

if __name__ =='__main__':
    pose_subscriber() 